src/plugin/kubearmor.go

```diff
@@ -590,13 +596,23 @@ func ConvertKubeArmorNetLogToKnoxNetLog(kaNwLogs []*pb.Log) []types.KnoxNetworkL
 				continue
 			}
 
-			destPod, destLabels, destNs := cluster.ExtractPodSvcInfoFromIP(ip, kalog.ClusterName)
+			if existingClustername != kalog.ClusterName {
+				_, services, _, pods, err = cluster.GetAllClusterResources(kalog.ClusterName)
```

Comment:
Actually we don't want to change the overall functionality as it may have some unforseen outcomes. The performance issue found in the pprof was particular to this function, that's why I only changed this function for now. 
Now regarding the performance improvement, here we are calling `GetAllClusterResources()` only once when we receive a unique clusterName from the logs. It's not called on each function, that's why am using the `existingClustername` variable.

Reply:
Okay got. It's fine.